raft/src/main/java/org/apache/kafka/raft/ReplicatedLog.java

```diff
@@ -103,6 +103,16 @@
      */
     void updateHighWatermark(LogOffsetMetadata offsetMetadata);
 
+    /**
+     * Flush the current log to disk.
+     */
+    void flush();
+
+    /**
+     * Get the last offset which has been flushed to disk.
+     */
+    long lastFlushedOffset();
```

Comment:
Is this function going to be used for non-testing code in the future?

Reply:
I think we will need it to implement the leader deferred fsync or any other optimization which allows the followers to read unflushed data from the leader. I decided to add it to the interface since we have a trivial implementation for it in `KafkaMetadataLog`. We can save it for later if you prefer.

Reply:
nah, I'm just curious to see what's in your future plan. Let's just keep it.